### PR TITLE
adguardian: init at 1.2.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Add package [`adguardian`](https://github.com/Lissy93/AdGuardian-Term), a terminal-based, real-time traffic monitoring and statistics for AdGuard Home.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 235178` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>adguardian</li>
  </ul>
</details>

File: pkgs/applications/networking/adguardian/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, fetchFromGitHub, rustPlatform }:
```
Comment (figsoda) [inline pkgs/applications/networking/adguardian/default.nix:1]:
```suggestion
{ lib, rustPlatform, fetchFromGitHub, stdenv, darwin }:
```

File: pkgs/applications/networking/adguardian/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Lissy93";
+    repo = "AdGuardian-Term";
+    rev = version;
+    hash = "sha256-UZIwVvBBBj82IxGuZPKaNc/UZI1DAh5/5ni3fjiRF4o=";
+  };
+
+  cargoSha256 = "sha256-MnhikzQNeCjK5bCjePw8k7pf7RR63k1eZjobENlQd94=";
+
```
Comment (figsoda) [inline pkgs/applications/networking/adguardian/default.nix:15]:
```suggestion

  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Security
  ];

```

**Review by figsoda — COMMENTED** [review]
this should make it build on darwin

File: pkgs/applications/networking/adguardian/default.nix (hunk 1 above)
Comment (GaetanLepage) [reply pkgs/applications/networking/adguardian/default.nix]: done !

File: pkgs/applications/networking/adguardian/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/applications/networking/adguardian/default.nix:20]: Done ! Let's now see if it builds.

**Comment by GaetanLepage** [comment]
Tests seem to be all right.
@figsoda have you the ability to try to build it on a mac ?

**Review by drupol — APPROVED** [review]

**Pull request merged**